crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -3232,6 +3235,50 @@ impl DiagnosticGuard<'_, '_> {
     }
 }
 
+impl DiagnosticGuard<'_, '_> {
+    fn resolve_applicability(&self, fix: &Fix) -> Applicability {
+        self.context
+            .settings
+            .fix_safety
+            .resolve_applicability(self.rule, fix.applicability())
+    }
+
+    /// Set the [`Fix`] used to fix the diagnostic.
+    #[inline]
+    pub(crate) fn set_fix(&mut self, fix: Fix) {
+        if !self.context.rules.should_fix(self.rule) {
+            self.fix = None;
+            return;
+        }
```

Comment:
Does this match Ruff's current behavior? Does that mean, we won't even **show** fixes when the fix is disabled. This seems wrong to me.

I'm not suggesting we should necessarily change this in this PR but we should at least open an issue for what I consider a bug. Instead, Ruff should keep the fix (but mark it in some way) and we then skip it when applying automatic fixes (and we don't show it in the LSP although I think even that would be fine).

Which makes me wonder. Should `should_fix == False` resolve to manual applicability?

Reply:
Let's create a follow up issue for this. I think it requires some design work on how we want to handle them in the LSP.